lib/jsonapi/serializer.ex

```diff
@@ -154,7 +154,7 @@ defmodule JSONAPI.Serializer do
 
     links =
       %{self: view.url_for(data, conn)}
-      |> Map.merge(view.links(data, conn))
+      |> Map.merge(view.links(data, conn, Map.get(doc, :meta, %{})))
```

Comment:
Is it possible to pull some of these out into captures and use them instead? It's awkward having a pipeline with a nested functions.